internal/provider/resources/resource_deployment.go

```diff
@@ -645,20 +645,6 @@ func (r *DeploymentResource) ValidateConfig(
 		return
 	}
 
-	// Need to do dynamic validation based on the executor and worker queues
```

Comment:
https://astronomer.slack.com/archives/C05V3UUHFAN/p1734363978465609

Seems that if we want to allow customers to set this dynamically, we cannot use this static checking